### DISCUSSION TITLE
Delete record set when all records within it are deleted

### DISCUSSION THREAD

**Discussion by bronwyncombs** [specify/specify7#5844] [Feature Requests]
**Describe**
If you delete all records in a record set, it will still exist in the record set list dialog with (0) records attached. This is the case when the records are deleted both from within and from outside the record set 

**To Reproduce**
1. Go to saved record set
2. Delete records (either internally or externally to record set)
3. Open record set list dialog from side bar
4. See error as empty record set

**Expected behavior**
Delete record set from list of record sets shown

**Screenshots**

<img width="520" alt="Screen Shot 2023-07-11 at 3 44 46 PM" src="https://github.com/specify/specify7/assets/135047322/c4f545d6-41a3-442c-baf4-9ec8973b156b">

**Desktop:**
 - OS: macOS
 - Browser: Chrome
 - Specify 7 Version: `v7.9-dev`

**Database Name:** If applicable
`fwri_6_1_23`

**Comment by maxpatiiuk** [comment]
Possible implementations:
1. If deleting a resource on the front-end in a record set, and it's the last resource, delete the record set. This will not work in all cases as resource might be deleted even when not in a context of a record set (or when looking at a different record set, as same resource can be in multiple record sets)
2. On the back-end, if resource is deleted, there is a code that also removes the resource from all record sets it might have been in - augment that code to also check if record set is now empty and should be deleted
